src/shared/orgListUtil.ts

```diff
@@ -256,38 +253,38 @@ export class OrgListUtil {
-    orgs: ExtendedAuthFields[]
-  ): Promise<ExtendedAuthFields[]> {
-    // Reduce the information to key value pairs with signupUsername as key
-    const contentMap: Dictionary<ExtendedScratchOrgInfo> = updatedContents.reduce((map, scratchOrgInfo) => {
-      if (scratchOrgInfo) {
-        map[scratchOrgInfo.SignupUsername] = scratchOrgInfo;
-      }
-      return map;
-    }, {});
-
-    for (const scratchOrgInfo of orgs) {
-      const updatedOrgInfo = contentMap[scratchOrgInfo.username];
-      if (updatedOrgInfo) {
-        scratchOrgInfo.signupUsername = updatedOrgInfo.SignupUsername;
-        scratchOrgInfo.createdBy = updatedOrgInfo.CreatedBy.Username;
-        scratchOrgInfo.createdDate = updatedOrgInfo.CreatedDate;
-        scratchOrgInfo.devHubOrgId = updatedOrgInfo.devHubOrgId;
-        scratchOrgInfo.attributes = updatedOrgInfo.attributes;
-        scratchOrgInfo.orgName = updatedOrgInfo.OrgName;
-        scratchOrgInfo.edition = updatedOrgInfo.Edition;
-        scratchOrgInfo.status = updatedOrgInfo.Status;
-        scratchOrgInfo.expirationDate = updatedOrgInfo.ExpirationDate;
-        scratchOrgInfo.isExpired = updatedOrgInfo.Status === 'Deleted';
-        scratchOrgInfo.namespace = updatedOrgInfo.Namespace;
-      } else {
-        // eslint-disable-next-line no-await-in-loop
-        const logger = await OrgListUtil.retrieveLogger();
-        logger.warn(`Can't find ${scratchOrgInfo.username} in the updated contents`);
-      }
+    orgs: ExtendedAuthFieldsScratch[]
+  ): Promise<FullyPopulatedScratchOrgFields[]> {
+    const contentMap = new Map(updatedContents.map((org) => [org.SignupUsername, org]));
+
+    const results = orgs.map((scratchOrgInfo): FullyPopulatedScratchOrgFields | string => {
+      const updatedOrgInfo = contentMap.get(scratchOrgInfo.username);
+      return updatedOrgInfo
+        ? {
+            ...scratchOrgInfo,
+            signupUsername: updatedOrgInfo.SignupUsername,
+            createdBy: updatedOrgInfo.CreatedBy.Username,
+            createdDate: updatedOrgInfo.CreatedDate,
+            devHubOrgId: updatedOrgInfo.devHubOrgId,
+            devHubId: updatedOrgInfo.devHubOrgId,
+            attributes: updatedOrgInfo.attributes,
+            orgName: updatedOrgInfo.OrgName,
+            edition: updatedOrgInfo.Edition,
+            status: updatedOrgInfo.Status,
+            expirationDate: updatedOrgInfo.ExpirationDate,
+            isExpired: updatedOrgInfo.Status === 'Deleted',
+            namespace: updatedOrgInfo.Namespace,
+          }
+        : `Can't find ${scratchOrgInfo.username} in the updated contents`;
+    });
+
+    const warnings = results.filter((result) => typeof result === 'string') as string[];
+    if (warnings.length) {
+      const logger = await OrgListUtil.retrieveLogger();
+      warnings.forEach((warning) => logger.warn(warning));
     }
 
-    return orgs;
+    return results.filter((result) => typeof result !== 'string') as FullyPopulatedScratchOrgFields[];
```

Comment:
``` suggestion
    const logger = await OrgListUtil.retrieveLogger();
    const contentMap = new Map(updatedContents.map((org) => [org.SignupUsername, org]));

    return orgs.filter((scratchOrgInfo) => {
      if (!contentMap.has(scratchOrgInfo.username)) {
        logger.warn(`Can't find ${scratchOrgInfo.username} in the updated contents`);
        return false;
      }
      return true;
    })
      .map((scratchOrgInfo): FullyPopulatedScratchOrgFields => {
        const updatedOrgInfo = contentMap.get(scratchOrgInfo.username)!;
        return {
          ...scratchOrgInfo,
          signupUsername: updatedOrgInfo.SignupUsername,
          createdBy: updatedOrgInfo.CreatedBy.Username,
          createdDate: updatedOrgInfo.CreatedDate,
          devHubOrgId: updatedOrgInfo.devHubOrgId,
          devHubId: updatedOrgInfo.devHubOrgId,
          attributes: updatedOrgInfo.attributes,
          orgName: updatedOrgInfo.OrgName,
          edition: updatedOrgInfo.Edition,
          status: updatedOrgInfo.Status,
          expirationDate: updatedOrgInfo.ExpirationDate,
          isExpired: updatedOrgInfo.Status === 'Deleted',
          namespace: updatedOrgInfo.Namespace
        };
      });
```

Reply:
The idea behind this retrieveLogger design: optimistically, we don't need to waste the (considerable!) time instantiating Loggers until we know we need one. 

Or maybe I'm missing the intent behind the suggestion